### PR TITLE
update adapter.js to version 3.1.0

### PR DESCRIPTION
**Pull request by xdumaine**
Major version bump here. @tgabi333 @fippo objections?

### CODE REVIEW

**Comment by tgabi333** [comment]
i was thinking about put webrtc-adapter to peer dependency in this project and the surroundings

**Comment by xdumaine** [comment]
@fippo is that what you were thinking?

File: rtcpeerconnection.js [hunk 1]
```diff
@@ -885,15 +885,17 @@ PeerConnection.prototype.createDataChannel = function (name, opts) {
...
             cb(null, res);
-        },
-        function (err) {
+        }, function (err) {
             cb(err);
-        }
-    );
+        });
+    } else {
+        return this.pc.getStats();
```
Comment (fippo) [inline rtcpeerconnection.js]: almost... i'd apply with arguments here for future compability!

**Comment by fippo** [comment]
ping? I think we can ignore the FF failure. Not sure if the tests were ever updated to geckodriver

**Comment by tgabi333** [comment]
What if we drop requiring adapter? As a library we should not define which
version of shim to use.

As we release a new major also we could also remove getStats and let the
shim do the work for users who use this library.

**Comment by fippo** [comment]
how much effort would it be to move everything under jingle+simplewebrtc to have adapter as a peer dependency?

i'd rather not have users call the getStats method directly on the peerconnection.

**Comment by xdumaine** [comment]
Sorry to leave this hanging. I'm ok with either option. Peer dependencies sound like a good idea. We're refactoring some code so that dependency management will be better as well so I'm ok with that.

**Comment by xdumaine** [comment]
Merging. Can do another major version bump to drop deps or move to peer deps.

**Pull request merged**